habitat-lab/habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -505,18 +527,63 @@ def generate_single_episode(self) -> Optional[RearrangeEpisode]:
             targ_sampler_name_to_obj_sampler_names[
                 sampler_name
             ] = targ_sampler_cfg["params"]["object_samplers"]
+        dataset_name = osp.basename(self.cfg.dataset_path).split(".", 1)[0]
```

Comment:
Break this into a separate method, the `generate_single_episode` method is getting too large and complex.